src/main/python/docs/source/guide/python_end_to_end_tut.rst

```diff
@@ -0,0 +1,561 @@
+The file can be found here:
+
+    - :doc:tests/examples/tutorials/neural_net_source.dml
+
+.. code-block:: python
+
+    FFN_package = sds.source(neural_net_src_path, "fnn", print_imported_methods=True))
+
+Step 3: Training the neural network
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Training a neural network in SystemDS using the train function is straightforward.
+The first two arguments are the training features and the target values we want to fit our model on.
+Then we need to set the hyperparameters of the model.
+We choose to train for 1 epoch with a batch size of 16 and a learning rate of 0.01, which are common parameters for neural networks.
+The seed argument ensures that running the code again yields the same results.
+
+.. code-block:: python
+
+    epochs = 1
+    batch_size = 16
+    learning_rate = 0.01
+    seed = 42
+
+    network = FFN_package.train(X, Y, epochs, batch_size, learning_rate, seed)
+
+Step 4: Saving the model
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For later usage, we can save the trained model.
+We only need to specify the name of our model and the file path.
+This call stores the weights and biases of our model.
+
+.. code-block:: python
+
+    FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
+
+Full Script
+~~~~~~~~~~~
+
+The complete script now can be seen here:
+
+.. code-block:: python
+
+    import numpy as np
+    from systemds.context import SystemDSContext
+
+    train_count = 32561
+    test_count = 16281
+
+    dataset_path_train = "adult/train_data.csv"
+    dataset_path_test = "adult/test_data.csv"
+    dataset_jspec = "adult/jspec.json"
+    preprocess_src_path = "preprocess.dml"
+    neural_net_src_path = "neural_net_source.dml"
+
+    sds = SystemDSContext()
+
+    SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+
+    F1 = sds.read(dataset_path_train, schema=SCHEMA)
+    F2 = sds.read(dataset_path_test,  schema=SCHEMA)
+
+    jspec = sds.read(dataset_jspec, data_type="scalar", value_type="string")
+    PREPROCESS_package = sds.source(preprocess_src_path, "preprocess", print_imported_methods=True)
+
+    X1 = F1.rbind(F2)
+    X1, M1 = X1.transform_encode(spec=jspec)
+
+    X = PREPROCESS_package.get_X(X1, 1, train_count)
+    Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+    Xt = PREPROCESS_package.get_X(X1, train_count, train_count+test_count)
+    Yt = PREPROCESS_package.get_Y(X1, train_count, train_count+test_count)
+
+    Yt = PREPROCESS_package.replace_value(Yt, 3.0, 1.0)
+    Yt = PREPROCESS_package.replace_value(Yt, 4.0, 2.0)
+
+    FFN_package = sds.source(neural_net_src_path, "fnn", print_imported_methods=True)
+
+    epochs = 1
+    batch_size = 16
+    learning_rate = 0.01
+    seed = 42
+
+    network = FFN_package.train(X, Y, epochs, batch_size, learning_rate, seed)
+
+    FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
```

Comment:
we can now save lists directly. therefore just simply call write, also when reading in the model again later, simply call read.